src/networking.c

```diff
@@ -438,18 +438,22 @@ static int read_client_data(conn_info *conn) {
  * of what to do.
  */
 static void handle_udp_message(aeEventLoop *loop, int fd, void *edata, int mask) {
-		statsite_networking *netconf = (statsite_networking *) edata;
+    struct iovec vectors[2];
+    int num_vectors;
+    ssize_t read_bytes;
+    statsite_networking *netconf = (statsite_networking *) edata;
 
-    while (1) {
-        // Get the associated connection struct
-        conn_info *conn = netconf->udp_client;
+    // Get the associated connection struct
+    conn_info *conn = netconf->udp_client;
 
-        // Clear the input buffer
-        circbuf_clear(&conn->input);
+    // Clear the input buffer
+    circbuf_clear(&conn->input);
 
+    // Loop until our buffer cannot hold another UDP packet (using a 1500
+    // byte MTU) or we cannot read another packet off the wire.  We do not
+    // read continuously off of the UDP socket to preserve timer execution.
+    while (circbuf_avail_buf(&conn->input) > 1500) {
```

Comment:
What would happen if for some reason the MTU would be different? Not sure if anyone is going to enable Jumbo frames on an Ethernet interface for Statsite, or maybe puts Statsite on a PPPoE connection... but you know, hardcoded assumptions and stuff. I myself think it's fine and putting a note in the docs would be fine as well, anyone else have anything to say about it?